src/kOS/Screen/ConnectivityInterpreter.cs

```diff
@@ -23,19 +23,32 @@ public class ConnectivityInterpreter : Interpreter, IUpdateObserver
         public ConnectivityInterpreter(SharedObjects shared) : base(shared)
         {
             Shared.UpdateHandler.AddObserver(this);
-            CreateProgressBarSubBuffer();
+            CreateProgressBarSubBuffer(this);
+            AddResizeNotifier(CreateProgressBarSubBuffer);
         }
 
-        private void CreateProgressBarSubBuffer()
+        ~ConnectivityInterpreter()
         {
-            progressBarSubBuffer = new SubBuffer();
+            RemoveResizeNotifier(CreateProgressBarSubBuffer);
+        }
```

Comment:
Addendum: Okay, actually it probably did work all along.  I put it back, with the explanatory comment about why it probably was fine all along.  But even so, I still added the hook removal to SharedObjects's Destroy work just to be explicit about it.  Now there's two spots that can both catch this and do the job.